File: appli-web/src/app/core/api/task-check-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { TaskCheck } from '@core/models/task-check.model';
import { API_BASE_URL } from './api.config';

@Injectable({ providedIn: 'root' })
export class TaskCheckApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/task-checks`;

  getAll(): Observable<TaskCheck[]> {
    return this.http.get<TaskCheck[]>(this.base);
  }

  upsert(c: TaskCheck): Observable<TaskCheck> {
    // Corps wire attendu par TaskCheckCreate (camelCase, sans userId).
    return this.http.put<TaskCheck>(`${this.base}/${c.uuid}`, {
      uuid: c.uuid,
      taskUUID: c.taskUUID,
      occurrenceDate: c.occurrenceDate,
      isChecked: c.isChecked,
      checkedAt: c.checkedAt ?? null,
      updatedAt: c.updatedAt,
    });
  }

  /** Upsert groupé `PUT /task-checks/bulk` (miroir upsertAll Android) — 1 requête pour N rows. */
  upsertAll(items: TaskCheck[]): Observable<TaskCheck[]> {
    return this.http.put<TaskCheck[]>(
      `${this.base}/bulk`,
      items.map((c) => ({
        uuid: c.uuid,
        taskUUID: c.taskUUID,
        occurrenceDate: c.occurrenceDate,
        isChecked: c.isChecked,
        checkedAt: c.checkedAt ?? null,
        updatedAt: c.updatedAt,
      })),
    );
  }

  delete(uuid: string): Observable<unknown> {
    return this.http.delete(`${this.base}/${uuid}`);
  }
}

File: appli-android/app/src/main/java/com/example/sportapp/core/network/TaskCheckApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.TaskCheck
import retrofit2.http.*

/**
 * Phase 0 (2026-05-12) : remplace RoutineTaskCheckApi. Routes /api/v1/task-checks/...
 */
interface TaskCheckApi {

    @GET("task-checks")
    suspend fun getAll(): List<TaskCheck>

    @GET("task-checks/{uuid}")
    suspend fun getByUUID(@Path("uuid") uuid: String): TaskCheck

    @PUT("task-checks/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body item: TaskCheck)

    @PUT("task-checks/bulk")
    suspend fun upsertAll(@Body items: List<TaskCheck>)

    @DELETE("task-checks/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}
